doc/CONFIGURE.md

```diff
@@ -0,0 +1,205 @@
+If the agent goes offline, e.g. due to a temporary network failure, Jenkins will periodically attempt to restart it.
+
+![](images/availability-as-much.png)
+
+### Take this agent online and offline at specific times
+In this mode, Jenkins will bring this agent online at the scheduled time(s), remaining online for a specified amount of time.
+If the agent goes offline while it is scheduled to be online, Jenkins will periodically attempt to restart it.
+
+After this agent has been online for the number of minutes specified in the Scheduled Uptime field, it will be taken offline. 
+If Keep online while builds are running is checked, and the agent is scheduled to be taken offline, 
+Jenkins will wait for any any builds that may be in progress to complete.
+
+![](images/availability-schedule.png)
+
+### Take this agent online when in demand, and offline when idle
+In this mode, Jenkins will bring this agent online if there is demand, i.e. there are queued builds which meet the following criteria:
+* They have been in the queue for at least the specified In demand delay time period
+* They can be executed by this agent (e.g. have a matching label expression)
+
+This agent will be taken offline if:
+* There are no active builds running on this agent
+* This agent has been idle for at least the specified Idle delay time period
+
+![](images/availability-on-demand.png)
+
+## Integration with SSH Credentials Plugin
+this plugin is now integrated with the [SSH Credentials Plugin](https://plugins.jenkins.io/ssh-credentials). 
+This changes how slaves are configured.
+The Node configuration is simplified, e.g. you now just have a Credentials drop down listing all the "Global" and 
+"System" scoped credentials.
+ 
+If you are upgrading from a previous 0.23 version, the plugin should try to inject any required SSH credentials in the 
+[Credentials Plugin](https://plugins.jenkins.io/credentials) using the credentials that were previously stored in each node's definition.
+
+![](images/ssh-credentials-select.png)
+
+To define credentials to use when connecting slaves you need to go to the `Jenkins/Manage Jenkins/Manage Credentials` screen 
+
+![](images/ssh-credentials-manage.png)
+
+Once on this screen you can add **SSH credentials**, either using a *Username & Password* or using a *Username & Private Key* 
+
+![](images/ssh-credentials-manage-detail.png)
+
+Credential scope controls where the credentials can be used:
+
+* System scope is only available for the root Jenkins instance (in other words Jenkins can use it to connect build nodes, but the credentials are not available to build jobs)
+* Global scope is available for the root Jenkins instance and any child items (in other words Jenkins can use it to connect build nodes, build jobs can use it for other SSH Credentials enabled plugins) 
+When you have a lot of different credentials it can be useful to put those credentials into credential domains, e.g.
```

Comment:
I'm not sure if this is specific to this plugin, but I found this description to be useful. Every time I've added SSH keys to Jenkins, I've always been confused by the point of the domain field.

Reply:
it is not specific to this plugin, it is SSH Credentials related stuff but it is good to document it here.